clarity/src/vm/types/signatures.rs

```diff
@@ -523,20 +622,20 @@ impl TypeSignature {
                     //   if both OkType and ErrType admit => admit
                     //   otherwise fail.
                     if other_inner_type.0.is_no_type() {
-                        my_inner_type.1.admits_type(&other_inner_type.1)
+                        my_inner_type.1.admits_type_v2_1(&other_inner_type.1)
                     } else if other_inner_type.1.is_no_type() {
-                        my_inner_type.0.admits_type(&other_inner_type.0)
+                        my_inner_type.0.admits_type_v2_1(&other_inner_type.0)
                     } else {
-                        Ok(my_inner_type.1.admits_type(&other_inner_type.1)?
-                            && my_inner_type.0.admits_type(&other_inner_type.0)?)
+                        Ok(my_inner_type.1.admits_type_v2_1(&other_inner_type.1)?
+                            && my_inner_type.0.admits_type_v2_1(&other_inner_type.0)?)
                     }
                 } else {
                     Ok(false)
                 }
             }
             TupleType(ref tuple_sig) => {
                 if let TupleType(ref other_tuple_sig) = &other {
-                    tuple_sig.admits(other_tuple_sig)
+                    tuple_sig.admits(&StacksEpochId::Epoch21, other_tuple_sig)
```

Comment:
Below, do we want a request to type-check an instance of `TraitReference` to be `unreachable!()`, like `CallableType` type-checking is unreachable in epoch 2.0?

Reply:
Yes, I added this in 2c2635f80da637a2a9f87db13a7e36042ea1dc95.